src/utils/matchingLines/index.ts

```diff
@@ -0,0 +1,82 @@
+import { CaseSensitivity, FilterLogic, MatchType } from "constants/enums";
+import { Filters } from "types/logs";
+
+/**
+ * `constructRegexToMatch` constructs an array of regex expressions that represents the current filters
+ * being applied.
+ * @param visibleFilters - filters which have visibility toggled on
+ * @returns an array of objects containing a regex expression and a boolean indicating if the match is exact or not
+ */
+export const constructRegexToMatch = (visibleFilters: Filters) => {
+  const regexToMatch: { regex: RegExp; isMatch: boolean }[] = [];
+
+  visibleFilters.forEach((f) => {
+    const isMatch = f.matchType === MatchType.Exact;
+    if (f.caseSensitive === CaseSensitivity.Sensitive) {
+      regexToMatch.push({
+        regex: new RegExp(f.name),
+        isMatch,
+      });
+    } else {
+      regexToMatch.push({
+        regex: new RegExp(f.name, "i"),
+        isMatch,
+      });
+    }
+  });
+
+  return regexToMatch;
+};
+
+/**
+ * `matchesFilters` tests if a given log line matches the applied filters.
+ * @param line - a log line
+ * @param regexToMatch - an array of objects representing the regex expressions that must be matched
+ * @param filterLogic - the filter mode, one of "AND" or "OR"
+ * @returns true if the line matches the filters, false otherwise
+ */
+export const matchesFilters = (
+  line: string,
+  regexToMatch: { regex: RegExp; isMatch: boolean }[],
+  filterLogic: FilterLogic
+) => {
+  if (filterLogic === FilterLogic.And) {
+    return regexToMatch.every(({ regex, isMatch }) =>
+      isMatch ? regex.test(line) : !regex.test(line)
+    );
+  }
+  return regexToMatch.some(({ regex, isMatch }) =>
+    isMatch ? regex.test(line) : !regex.test(line)
+  );
+};
+
+/**
+ * 'getMatchingLines' determines which log lines match the applied filters. This function returns undefined if no
+ * filters have been applied.
+ * @param logLines - list of strings representing the log lines
+ * @param filters - list of filters being applied
+ * @param filterLogic - specifies whether to use AND or OR filtering
+ * @returns a set of numbers representing the matched lines, or undefined
+ */
+export const getMatchingLines = (
+  logLines: string[],
+  filters: Filters,
+  filterLogic: FilterLogic
+) => {
+  const set = new Set<number>();
+
+  const visibleFilters = filters.filter((f) => f.visible !== false);
```

Comment:
It doesn't look like the filter rules are being obeyed. For example i have one filter that is hidden and one that is visible and i would expect to see the rules from the visible one but instead it seems to be showing no lines.
https://user-images.githubusercontent.com/4605522/197627122-32480cb4-1633-4d2e-8a38-4fba8d5eb89e.mov

Reply:
I think you may have attached the wrong video haha! But yes I was passing the wrong thing down to other functions. I added a test for this 🙏